wolfssl/wolfcrypt/sp_int.h

```diff
@@ -178,6 +178,15 @@ extern "C" {
 #define WOLFSSL_SP_DIV_WORD_HALF
 #endif
 
+/* Detect Cortex M3 (no UMAAL)*/
+#ifdef WOLFSSL_SP_ARM_CORTEX_M_ASM
+    #if defined(__ARM_ARCH) && (__ARM_ARCH < 8) && defined(__ARM_ARCH_7M__)
```

Comment:
Should this be:
#if defined(__ARM_ARCH) && (__ARM_ARCH < 8) || defined(__ARM_ARCH_7M__)

Reply:
No, that would disable it for M4. It could be just `#if defined(ARM_ARCH_7M)`.

These are set exclusively (only one or the other) for each platform:
```
M3: #define __ARM_ARCH_7M__ 1
M4: #define __ARM_ARCH_7EM__ 1
```

Reply:
If you want I could rewrite to:

```
#if defined(WOLFSSL_SP_ARM_CORTEX_M_ASM) && defined(__ARM_ARCH_7M__)
    #undef  WOLFSSL_SP_NO_UMAAL
    #define WOLFSSL_SP_NO_UMAAL
#endif
```

Reply:
OK make it just __ARM_ARCH_7M__ like in the your last comment.